src/test/java/org/broadinstitute/hellbender/engine/dataflow/transforms/PairReadsWithRefBasesUnitTest.java

```diff
@@ -54,4 +61,45 @@ public void refBasesTest(List<KV<ReferenceBases, Iterable<GATKRead>>> kvRefBases
         p.run();
     }
 
+    @DataProvider(name = "PairReadsWithRefBasesWithCustomWindowFunctionTestData")
+    public Object[][] pairReadsWithRefBasesWithCustomWindowFunctionTestData() {
+        final byte[] bases = "AGCCTTTCGAACTGAGCCCGTTCCTGGGGTTATACCCGGCTTTTGGCGCT".getBytes();
+        final ReferenceBases refBases = new ReferenceBases(bases, new SimpleInterval("1", 1, 50));
+
+        final List<Object[]> testCases = new ArrayList<>();
+        for ( final Class<?> readImplementation : Arrays.asList(SAMRecord.class, Read.class) ) {
+            // Test case layout: Ref bases + read, reference window function to apply, expected ref bases for read, expected ref interval for read
+            testCases.add(new Object[]{ KV.of(refBases, Arrays.asList(makeRead("1", 1, 10, 0, readImplementation))), RefWindowFunctions.IDENTITY_FUNCTION, "AGCCTTTCGA".getBytes(), new SimpleInterval("1", 1, 10) });
```

Comment:
I think adding a short comment by each test case would make them easier to get (like going off the end or beginning, etc.)